crates/omnix-health/src/lib.rs

```diff
@@ -75,22 +77,36 @@ impl NixHealth {
     #[instrument(skip_all)]
     pub fn run_all_checks(
         &self,
-        nix_info: &nix_rs::info::NixInfo,
+        nix_info: &NixInfo,
         flake_url: Option<FlakeUrl>,
-    ) -> Vec<traits::Check> {
+    ) -> HashMap<String, Check> {
         self.into_iter()
             .flat_map(|c| c.check(nix_info, flake_url.as_ref()))
             .collect()
     }
 
-    pub async fn print_report_returning_exit_code(checks: &[traits::Check]) -> anyhow::Result<i32> {
+    pub async fn print_json_report_returning_exit_code(checks_map: &HashMap<String, Check>) -> anyhow::Result<i32> {
```

Comment:
Lot of overlap between `print_json_report_returning_exit_code` and `print_report_returning_exit_code`, should we factor out the commonalities?

Reply:
Yes, we can make it take a `json_only` argument.

Reply:
Also, can make you this print JSON in a single line (which would be useful in bash scripts)

Reply:
https://github.com/juspay/omnix/pull/379/commits/415f3e42411cdac7d058732efed213ca7224186c (json_only)

https://github.com/juspay/omnix/pull/379/commits/c678a6c8b97f5b9957dae7882bce119b4614ccf0 (single line json)